src/Widgets/ToolbarButton.vala

```diff
@@ -100,15 +99,67 @@ public class ENotes.ToolbarButton : Gtk.Button {
                 } else if (type == 2) {
                     plugin.request_string ("");
                 } else {
-                    code_buffer.insert_at_cursor (first_half, -1);
-                    int end_first_half_pos = code_buffer.cursor_position;
-                    code_buffer.insert_at_cursor (second_half, -1);
-
                     Gtk.TextIter cursor_position;
-                    code_buffer.get_iter_at_offset (out cursor_position, end_first_half_pos);
-                    code_buffer.place_cursor (cursor_position);
+                    code_buffer.get_iter_at_offset (out cursor_position, code_buffer.cursor_position);
+
+                    if (cursor_position.inside_word () || cursor_position.ends_word ()) {
+                        // gets word the cursor is currently on and modify it
+                        start = end = cursor_position;
+                        identify_word (ref start, ref end, first_half, second_half);
+                        var text = start.get_text (end);
+                        var wrapped_text = WordWrapper.get_wrapped_text (text, first_half, second_half);
+                        changes_text (ref code_buffer, start, end, wrapped_text);
+                    } else {
+                        // prints the wrapping text and put cursor in the middle 
+                        code_buffer.insert_at_cursor (first_half + second_half, -1);
+                        code_buffer.get_iter_at_offset (out cursor_position, code_buffer.cursor_position);
+                        cursor_position.backward_chars (second_half.length);
+                        code_buffer.place_cursor (cursor_position);
+                    }
                 }
             }
         });
     }
+
+    /**
+     * Replaces the content from  iter start to iter end with the informed text
+     */
+    private void changes_text (ref Gtk.SourceBuffer buffer, Gtk.TextIter start, Gtk.TextIter end, string text) {
+        code_buffer.@delete (ref start, ref end);
+        code_buffer.insert_at_cursor (text, -1);
+    }
+
+    /**
+     * Adjusts iterators to surround a word and its wrapper, if there is one
+     */
+    private static void identify_word (ref Gtk.TextIter start, ref Gtk.TextIter end, string first_half, string second_half) {
+            if (!start.starts_word ()) {
+                start.backward_word_start () ;
+            }
+            if (!end.ends_word ()) {
+                end.forward_word_end ();
+            }
+            if (iter_starts_after (start, first_half) && iter_is_followed_by (end, second_half)) {
+                start.backward_chars (first_half.length);
+                end.forward_chars (second_half.length);
+            }
```

Comment:
Check the white space, you got 8 spaces between the if blocks and the indentation of the function